### ISSUE TITLE
Bind backend crash when invalid named.conf provided

### ISSUE THREAD

**Issue by cmouse** [PowerDNS/pdns#2334]
With named.conf

```
options {
  directory /home/cmouse/pdns/etc;
};

zone "unit.test" {
  type master;
  file "db.unit.test";
};
```

You get signal 6

```
Mar 08 12:32:13 Creating backend connection for TCP
Mar 08 12:32:13 Error parsing bind configuration: Error in bind configuration '/home/cmouse/pdns/etc/named.conf' on line 2: syntax error
Unable to close down sqlite connection: 5
Mar 08 12:32:13 Got a signal 6, attempting to print trace: 
Mar 08 12:32:13 ../sbin/pdns_server(+0x3d02f6) [0x7f7a8b7812f6]
Mar 08 12:32:13 /lib/x86_64-linux-gnu/libc.so.6(+0x36d40) [0x7f7a88c57d40]
Mar 08 12:32:13 /lib/x86_64-linux-gnu/libc.so.6(gsignal+0x39) [0x7f7a88c57cc9]
Mar 08 12:32:13 /lib/x86_64-linux-gnu/libc.so.6(abort+0x148) [0x7f7a88c5b0d8]
Mar 08 12:32:13 ../sbin/pdns_server(_ZN8SSQLite3D1Ev+0x8b) [0x7f7a8b82cec5]
Mar 08 12:32:13 ../sbin/pdns_server(_ZN8SSQLite3D0Ev+0x18) [0x7f7a8b82cf0c]
Mar 08 12:32:13 ../sbin/pdns_server(_ZN5boost14checked_deleteI8SSQLite3EEvPT_+0x2a) [0x7f7a8b892dca]
Mar 08 12:32:13 ../sbin/pdns_server(_ZN5boost6detail17sp_counted_impl_pI8SSQLite3E7disposeEv+0x1c) [0x7f7a8b892e50]
Mar 08 12:32:13 ../sbin/pdns_server(_ZN5boost6detail15sp_counted_base7releaseEv+0x42) [0x7f7a8b6aec52]
Mar 08 12:32:13 ../sbin/pdns_server(_ZN5boost6detail12shared_countD1Ev+0x27) [0x7f7a8b6aece1]
Mar 08 12:32:13 ../sbin/pdns_server(_ZN5boost10shared_ptrI8SSQLite3ED1Ev+0x1c) [0x7f7a8b885c16]
Mar 08 12:32:13 ../sbin/pdns_server(_ZN12Bind2BackendC1ERKSsb+0x9cf) [0x7f7a8b87da6f]
Mar 08 12:32:13 ../sbin/pdns_server(_ZN12Bind2Factory4makeERKSs+0x34) [0x7f7a8b88674e]
Mar 08 12:32:13 ../sbin/pdns_server(_ZN17BackendMakerClass3allEb+0x194) [0x7f7a8b6c11dc]
Mar 08 12:32:13 ../sbin/pdns_server(_ZN12UeberBackendC2ERKSs+0x19c) [0x7f7a8b7e7a2e]
Mar 08 12:32:13 ../sbin/pdns_server(_ZN13PacketHandlerC1Ev+0x38) [0x7f7a8b7651da]
Mar 08 12:32:13 ../sbin/pdns_server(_ZN13TCPNameserver2goEv+0xe1) [0x7f7a8b7d4113]
Mar 08 12:32:13 ../sbin/pdns_server(_Z10mainthreadv+0xf3a) [0x7f7a8b69a952]
Mar 08 12:32:13 ../sbin/pdns_server(main+0x312c) [0x7f7a8b784643]
Mar 08 12:32:13 /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf5) [0x7f7a88c42ec5]
Aborted (core dumped)
```

**Comment by cmouse** [comment]
Bind backend also crashes if bind-config is defined, bind-dnssec-db is defined, but config-file is missing

```
(gdb) bt
#0  0xb6b39b98 in raise () from /usr/lib/libc.so.6
#1  0xb6b3afac in abort () from /usr/lib/libc.so.6
#2  0x7f75fac8 in SSQLite3::~SSQLite3 (this=<optimized out>, __in_chrg=<optimized out>) at ssqlite3.cc:171
#3  0x7f75faec in SSQLite3::~SSQLite3 (this=0x7f8806a8, __in_chrg=<optimized out>) at ssqlite3.cc:177
#4  0x7f60deac in std::_Sp_counted_base<(__gnu_cxx::_Lock_policy)1>::_M_release (this=0x7f880dc0) at /usr/include/c++/5.2.0/bits/shared_ptr_base.h:150
#5  0x7f779c78 in std::__shared_count<(__gnu_cxx::_Lock_policy)1>::~__shared_count (this=0x7f880594, __in_chrg=<optimized out>) at /usr/include/c++/5.2.0/bits/shared_ptr_base.h:659
#6  std::__shared_ptr<SSQLite3, (__gnu_cxx::_Lock_policy)1>::~__shared_ptr (this=0x7f880590, __in_chrg=<optimized out>) at /usr/include/c++/5.2.0/bits/shared_ptr_base.h:925
#7  Bind2Backend::Bind2Backend (this=0x7f880588, suffix=..., loadZones=<optimized out>) at /usr/include/c++/5.2.0/bits/shared_ptr.h:93
#8  0x7f77d564 in Bind2Factory::make (this=<optimized out>, suffix="") at bindbackend2.cc:1366
#9  0x7f61cb40 in BackendMakerClass::all (this=this@entry=0x7f82c4fc <BackendMakers()::bmc>, metadataOnly=<optimized out>) at dnsbackend.cc:193
#10 0x7f721674 in UeberBackend::UeberBackend (this=0x7f87fdb4, pname="pdns") at ueberbackend.cc:414
#11 0x7f6b5988 in PacketHandler::PacketHandler (this=0x7f87fda8) at packethandler.cc:64
#12 0x7f70ea30 in TCPNameserver::go (this=0x7f87cde0) at tcpreceiver.cc:76
#13 0x7f5fc3e4 in mainthread () at common_startup.cc:523
#14 0x7f5c2654 in main (argc=-1090519469, argv=0xbefffb40) at receiver.cc:606
```

**Comment by Habbie** [comment]
closed by #2966
